code/game/objects/items/devices/traitordevices.dm

```diff
@@ -168,3 +168,203 @@ effective or pretty fucking useless.
+			new/obj/effect/temp_visual/teleport_abductor/syndi_teleporter(destination)
+		else if (EMP_D == FALSE && !(bagholding.len && !flawless)) // This is where the fun begins
+			var/direction = get_dir(user, destination)
+			panic_teleport(user, destination, direction)
+		else // Emp activated? Bag of holding? No saving throw for you
+			get_fragged(user, destination)
+	else
+		to_chat(C, "<span class='danger'>The [src] will not work here!</span>")
+
+/obj/item/teleporter/proc/tile_check(turf/T)
+	if(istype(T, /turf/simulated/floor) || istype(T, /turf/space) || istype(T, /turf/simulated/shuttle/floor) || istype(T, /turf/simulated/shuttle/floor4) || istype(T, /turf/simulated/shuttle/plating))
+		return TRUE
+
+/obj/item/teleporter/proc/panic_teleport(mob/user, turf/destination, direction = NORTH)
+	var/saving_throw
+	switch(direction)
+		if(NORTH || SOUTH)
+			if(prob(50))
+				saving_throw = EAST
+			else
+				saving_throw = WEST
+		if(EAST || WEST)
+			if(prob(50))
+				saving_throw = NORTH
+			else
+				saving_throw = SOUTH
+		else
+			saving_throw = NORTH // just in case
+
+	var/mob/living/carbon/C = user
+	var/turf/mobloc = get_turf(C)
+	var/list/turfs = list()
+	var/found_turf = FALSE
+	for(var/turf/T in range(destination, saving_throw_distance))
+		if(get_dir(destination, T) != saving_throw)
+			continue
+		if(T.x > world.maxx-saving_throw_distance || T.x < saving_throw_distance)
+			continue	//putting them at the edge is dumb
+		if(T.y > world.maxy-saving_throw_distance || T.y < saving_throw_distance)
+			continue
+		if(!tile_check(T))
+			continue // We are only looking for safe tiles on the saving throw, since we are nice
+		turfs += T
+		found_turf = TRUE
+
+	if(found_turf)
+		var/turf/new_destination = pick(turfs)
+		var/turf/fragging_location = new_destination
+		telefrag(fragging_location, user)
+		C.forceMove(new_destination)
+		playsound(mobloc, "sparks", 50, TRUE)
+		new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(mobloc)
+		new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(new_destination)
+		playsound(new_destination, "sparks", 50, TRUE)
+	else //We tried to save. We failed. Death time.
+		get_fragged(user, destination)
+
+
+/obj/item/teleporter/proc/get_fragged(mob/user, turf/destination)
+	var/turf/mobloc = get_turf(user)
+	user.forceMove(destination)
+	playsound(mobloc, "sparks", 50, TRUE)
+	new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(mobloc)
+	new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(destination)
+	playsound(destination, "sparks", 50, TRUE)
+	playsound(destination, "sound/magic/disintegrate.ogg", 50, TRUE)
+	destination.ex_act(rand(1,2))
+	for(var/obj/item/W in user)
+		if(istype(W, /obj/item/organ)|| istype(W, /obj/item/implant))
+			continue
+		if(!user.unEquip(W))
+			qdel(W)
+	to_chat(user, "<span class='userdanger'><font size=3>You teleport into the wall, the teleporter tries to save you, but--</font></span>")
```

Comment:
`biggerdanger` may be a better class here, then you dont have to stack tags.

Reply:
Done